src/cargo/ops/cargo_package.rs

```diff
@@ -405,6 +423,31 @@ fn warn_on_nonexistent_file(
     ))
 }
 
+fn error_custom_build_file_not_in_package(
+    pkg: &Package,
+    path: &Path,
+    target: &Target,
+) -> CargoResult<Vec<ArchiveFile>> {
+    let tip = {
+        let description_name = target.description_named();
+        if path.is_file() {
+            format!("the source file of `{description_name}` doesn't appear to be a path inside of the package.\n\
```

Comment:
Just one tiny suggestion and we're there!

It seems odd to have backticks here. Could we remove those?

```suggestion
            format!("the source file of {description_name} doesn't appear to be a path inside of the package.\n\
```

Reply:
It seems good to me to remove the backticks.